src/Connector.AzureDataLake/Connector/AzureDataLakeConnector.cs

```diff
@@ -135,9 +325,35 @@ public override Task<IAsyncEnumerable<ConnectorLatestEntityPersistInfo>> GetLate
 
         public override async Task<ConnectionVerificationResult> VerifyConnection(ExecutionContext executionContext, IReadOnlyDictionary<string, object> config)
         {
-            await _client.EnsureDataLakeDirectoryExist(new AzureDataLakeConnectorJobData(config.ToDictionary(x => x.Key, x => x.Value)));
+            try
+            {
+                var jobData = await AzureDataLakeConnectorJobData.Create(executionContext, config.ToDictionary(config => config.Key, config => config.Value));
+                await _client.EnsureDataLakeDirectoryExist(jobData);
 
-            return new ConnectionVerificationResult(true);
+                if (jobData.IsStreamCacheEnabled)
+                {
+                    if (string.IsNullOrWhiteSpace(jobData.StreamCacheConnectionString))
+                    {
+                        return new ConnectionVerificationResult(false, $"Buffer connection string must be valid when buffer is enabled.");
+                    }
+                    await using var connection = new SqlConnection(jobData.StreamCacheConnectionString);
+                    await connection.OpenAsync();
+                    await using var connectionAndTransaction = await connection.BeginTransactionAsync();
+                    var connectionIsOpen = connectionAndTransaction.Connection.State == ConnectionState.Open;
+                    await connectionAndTransaction.DisposeAsync();
```

Comment:
We should try and create a temporal table here to verify the DB supports them and that the user has access to create them